OKTA-346990 update readme(s) to include extended configuration information (#158)
okta/okta-aspnet: docs/aspnetcore-mvc.md

}
```

## Configuration for cloud services or load balancers

If you plan to deploy your application to a cloud service, you may need to do additional customization. Cloud services usually create similar environments in which applications run inside a Docker container without using HTTPS. The front host or load balancer that the user has direct access to may or may not use HTTPS by-default. 

Cloud services are vary in configuration, but general rules are:

* Make sure Load Balancer or front host uses HTTPS. 

This may be the default or may require some amount of fiddling as in case of AWS.

> * Your Application should be aware of HTTPS scheme used by user despite the App itself is using HTTP. 

Reviewer: I'm not clear what this sentence is trying to communicate.  **Application** should not be capitalized and **App** should be spelled out and not capitalized.

> [Microsoft.AspNetCore.Authentication.BuildRedirectUri](https://github.com/aspnet/Security/blob/26d27d871b7992022c082dc207e3d126e1d9d278/src/Microsoft.AspNetCore.Authentication/AuthenticationHandler.cs#L117) function relies on request scheme and will not work properly when front host uses HTTPS but the application is running over HTTP. 

Reviewer: I would use "**load balancer**" instead of front host, here and all other places in this section.

```suggestion
[Microsoft.AspNetCore.Authentication.BuildRedirectUri](https://github.com/aspnet/Security/blob/26d27d871b7992022c082dc207e3d126e1d9d278/src/Microsoft.AspNetCore.Authentication/AuthenticationHandler.cs#L117) function relies on request scheme and will not work properly when load balancer uses HTTPS but the application is running over HTTP. 
```

> Please check this [Microsoft documentation](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer) on how to configure your application to work in such environments.

Reviewer: ```suggestion
For information on how to configure your application in a proxied, load balanced environment, see [Configure ASP.NET Core to work with proxy servers and load balancers](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer).
```

> Azure environment with Front Door load balancer should work properly after forwarded headers configuration is added like so:

Reviewer: I don't think **Front Door** should be capitalized here.  The term **front door load balancer** should be defined in the first paragraph of this section; you may want to reduce the number of names used for this (front host, load balancer, front door load balancer).  On line 92 above, introduce the names/concepts, define them clearly, then use one identifier in the remainder of the document.

```suggestion
Set forwarded headers as follows to properly configure your Azure load balancer.
```

`Startup.ConfigureServices:`
```csharp
     services.Configure<ForwardedHeadersOptions>(options =>
     {
         options.ForwardedHeaders =
             ForwardedHeaders.XForwardedHost | ForwardedHeaders.XForwardedProto;
     });
```

> Following is found to be working configuration for AWS or Google Cloud with App Engine Flex:

Reviewer: ```suggestion
Use code similar to the following to properly configure a load balanced `AWS` or `Google Cloud with App Engine Flex` environment.
```

`Startup.Configure:`
```csharp
     app.Use((context, next) =>
     {
         var xProtoHeaders = context.Request.Headers["X-Forwarded-Proto"];
         if (xProtoHeaders.Count > 0 && xProtoHeaders[0].ToLower() == "https")
             context.Request.Scheme = "https";
         return next();
     });
```

> Don't use `app.UseHttpsRedirection();` as it may cause infinite redirection loop on client.

Reviewer: ```suggestion
Don't use `app.UseHttpsRedirection();` as it may cause a client infinite redirection loop.
```

* Do additional configuration if needed. 

> Because of Docker specifics on some cloud services authentication may fail with the message `Unable to unprotect the message.State.` This most likely means you need to configure Data Protection storage. Follow [these instructions](https://cloud.google.com/appengine/docs/flexible/dotnet/application-security#aspnet_core_data_protection_provider) to do this for Google Cloud.

Reviewer: Text of a link should be the title of the target document.  
```suggestion
Because of Docker specifics on some cloud services authentication may fail with the message `Unable to unprotect the message.State.` This most likely means you need to configure Data Protection storage.  For Google Cloud, see [ASP.Net Core data protection provider](https://cloud.google.com/appengine/docs/flexible/dotnet/application-security#aspnet_core_data_protection_provider).
```
